src/components/FormModal/FormModal.js

```diff
@@ -92,27 +98,100 @@ export default function FormModal({ onCancelAdding, id, name }) {
   }
 
   function handleMorningClicked(name) {
-    const newMorningChecked = weekRoutine.days.map((day) => {
-      if (day.name === name) {
-        day.morning = !day.morning;
-        return day;
-      } else {
-        return day;
-      }
+    const i = getIndexForWeekday(name);
+    const checkedMornings = routineData.filter(
+      (product) =>
+        product.days[i].name === name && product.days[i].morning === true
+    );
+
+    const routineIDs = checkedMornings.map((product) => product.id);
+    const intersection = routineIDs.filter((id) => conflicts.includes(id));
+
+    const conflictName = intersection.map((id) => {
+      const conflictProduct = getProductById(id, products);
+      return conflictProduct.name;
     });
-    setWeekRoutine({ id: id, days: newMorningChecked, date: weekRoutine.date });
+
+    if (intersection.length > 0) {
+      alert(
+        `You are already using ${conflictName} on ${name} morning. These two products have conflicting ingredients`
+      );
+    } else {
+      const newMorningChecked = weekRoutine.days.map((day) => {
+        if (day.name === name) {
+          day.morning = !day.morning;
+          return day;
+        } else {
+          return day;
+        }
+      });
+      setWeekRoutine({
+        id: id,
+        days: newMorningChecked,
+        date: weekRoutine.date,
+      });
+    }
   }
 
   function handleEveningClicked(name) {
-    const newEveningChecked = weekRoutine.days.map((day) => {
-      if (day.name === name) {
-        day.evening = !day.evening;
-        return day;
-      } else {
-        return day;
-      }
+    const i = getIndexForWeekday(name);
+    const checkedEvenings = routineData.filter(
+      (product) =>
+        product.days[i].name === name && product.days[i].evening === true
+    );
+
+    const routineIDs = checkedEvenings.map((product) => product.id);
+    const intersection = routineIDs.filter((id) => conflicts.includes(id));
+
+    const conflictName = intersection.map((id) => {
+      const conflictProduct = getProductById(id, products);
+      return conflictProduct.name;
+    });
+
+    if (intersection.length > 0) {
+      alert(
+        `You are already using ${conflictName} on ${name} evening. These two products have conflicting ingredients`
+      );
+    } else {
+      const newEveningChecked = weekRoutine.days.map((day) => {
+        if (day.name === name) {
+          day.evening = !day.evening;
+          return day;
+        } else {
+          return day;
+        }
+      });
+      setWeekRoutine({
+        id: id,
+        days: newEveningChecked,
+        date: weekRoutine.date,
+      });
+    }
+  }
+
+  function handleModalFormSubmit(event) {
+    event.preventDefault();
+
+    const noDaysChecked = weekRoutine.days.map((day) => {
```

Comment:
The variable name is hard to understand. First I thought, in here are the days, which are not checked, but after understanding the code, it seems, that the variable holds a list of booleans. Maybe a good start would be to start with a `is` in the variable word. And maybe end with the word `list`.

```suggestion
    const isnNoDaysCheckedList = weekRoutine.days.map((day) => {
```

Read more [here](https://stackoverflow.com/questions/1227998/naming-conventions-what-to-name-a-boolean-variable) about it.